random/module.py

```diff
@@ -0,0 +1,217 @@
+            author=ctx.author,
+            footer="thedogapi.com",
+        )
+        embed.set_image(url=json_response[0]["url"])
+
+        await ctx.reply(embed=embed)
+
+    @commands.cooldown(rate=5, per=60, type=commands.BucketType.channel)
+    @commands.command()
+    async def xkcd(self, ctx, number: int = None):
+        """Get random xkcd comics
+
+        Arguments
+        ---------
+        number: Comics number. Omit to get random one.
+        """
+        # get maximal
+        async with aiohttp.ClientSession() as session:
+            async with session.get("https://xkcd.com/info.0.json") as response:
+                fetched = await response.json()
+
+                # get random
+                if number is None or number < 1 or number > fetched["num"]:
+                    number: int = random.randint(1, fetched["num"])
+                # fetch requested
+                if number != fetched["num"]:
+                    async with session.get(f"https://xkcd.com/{number}/info.0.json") as response:
+                        fetched = await response.json()
+
+        embed: discord.Embed = utils.Discord.create_embed(
+            author=ctx.author,
+            title=fetched["title"],
+            description="_" + fetched["alt"][:2046] + "_",
+            footer="xkcd.com",
+        )
+
+        embed.add_field(
+            name=(
+                f"{fetched['year']}"
+                f"-{str(fetched['month']).zfill(2)}"
+                f"-{str(fetched['day']).zfill(2)}"
+            ),
+            value=(
+                    f"https://xkcd.com/{number}\n"
+                    + f"https://www.explainxkcd.com/wiki/index.php/{number}"
+            ),
+            inline=False,
+        )
+        embed.set_image(url=fetched["img"])
+
+        await ctx.reply(embed=embed)
+
+    @commands.cooldown(rate=5, per=60, type=commands.BucketType.channel)
+    @commands.command()
+    async def dadjoke(self, ctx, *, keyword: Optional[str] = None):
+        """Get random dad joke
+
+        Arguments
+        ---------
+        keyword: search for a certain keyword in a joke
+        """
+        if keyword is not None and ("&" in keyword or "?" in keyword):
+            return await ctx.reply(_(ctx, "I didn't find a joke like that."))
+
+        params: Dict[str, str] = {"limit": "30"}
+        url: str = "https://icanhazdadjoke.com"
+        if keyword is not None:
+            params["term"] = keyword
+            url += "/search"
+        headers: Dict[str, str] = {"Accept": "application/json"}
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url, headers=headers, params=params) as response:
+                fetched = await response.json()
+
+        if keyword is not None:
+            res = fetched["results"]
+            if len(res) == 0:
+                return await ctx.reply(_(ctx, "I didn't find a joke like that."))
+            result = random.choice(res)
```

Comment:
One enhancement request. Can you underline the keyword in the joke, if there is one?
```py
# something like
joke.replace(keyword, f"__{keyword}__")
```
This should be a separate commit, because it differs from the original cog.